src/common/stringutils.cpp

```diff
@@ -121,43 +142,77 @@ unsigned int StrUtils::Utf8CharToUnicode(const std::string &ch)
 {
     if (ch.empty())
         return 0;
-
+    //  memo UTF8:
+    // 1 byte  : 00000000 -- 0000007F:  0xxxxxxx
+    // 2 bytes : 00000080 -- 000007FF:  110xxxxx 10xxxxxx
+    // 3 bytes : 00000800 -- 0000FFFF:  1110xxxx 10xxxxxx 10xxxxxx
+    // 4 bytes : 00010000 -- 001FFFFF:  11110xxx 10xxxxxx 10xxxxxx 10xxxxxx
     unsigned int result = 0;
     if ((ch[0] & 0x80) == 0)
     {
         if (ch.size() == 1)
-        result = static_cast<unsigned int>(ch[0]);
+        {
+            result = static_cast<unsigned int>(ch[0]);
+            //GetLogger()->Trace("StrUtils::Utf8CharToUnicode 1: <%s> %d\n",ch.c_str(),result);
+        }
+        else
+            GetLogger()->Warn("StrUtils::Utf8CharToUnicode Bad UTF8 1 char input ? : <%s>\n",ch.c_str());
     }
-    else if ((ch[0] & 0xC0) == 0xC0)
+    else if ((ch[0] & 0xE0) == 0xC0)
     {
         if (ch.size() == 2)
         {
             unsigned int ch1 = (ch[0] & 0x1F) << 6;
             unsigned int ch2 = (ch[1] & 0x3F);
             result = ch1 | ch2;
+            GetLogger()->Trace("StrUtils::Utf8CharToUnicode 2: <%s> %d\n",ch.c_str(),result);
         }
+        else
+            GetLogger()->Warn("StrUtils::Utf8CharToUnicode Bad UTF8 2 char input ? : <%s>\n",ch.c_str());
```

Comment:
Also, what is with the formatting? Why spaces around colons? Why no space after comma?